lib/tasks/server/middleware/history-support/index.js

```diff
@@ -29,7 +29,9 @@ HistorySupportAddon.prototype.serverMiddleware = function(config) {
 
         if (hasHTMLHeader && isForBaseURL && req.method === 'GET') {
           var assetPath = req.path.slice(baseURL.length);
-          if (!fs.existsSync(path.join(results.directory, assetPath))) {
+          var isFile = false;
+          try { isFile = fs.statSync(path.join(results.directory, assetPath)).isFile(); } catch (err) { }
+          if (!isFile) {
```

Comment:
If you need Broccoli to call `next()` on non-GET (or some other requests like that) to make that possible, I'm happy to merge a PR.